client/components/d3chart/d3-base/index.js

```diff
@@ -26,70 +26,34 @@ import './style.scss';
  * phase' (i.e. in 'componentDidMount' and 'componentDidUpdate' methods).
  */
 export default class D3Base extends Component {
-	static propTypes = {
-		className: PropTypes.string,
-		data: PropTypes.any, // required to detect changes in data
-		drawChart: PropTypes.func.isRequired,
-		getParams: PropTypes.func.isRequired,
-		type: PropTypes.string,
-	};
-
-	state = {
-		data: null,
-		params: null,
-		drawChart: null,
-		getParams: null,
-		type: null,
-	};
-
-	chartRef = createRef();
-
-	static getDerivedStateFromProps( nextProps, prevState ) {
-		let state = {};
-
-		if ( ! isEqual( nextProps.data, prevState.data ) ) {
-			state = { ...state, data: nextProps.data };
-		}
-
-		if ( ! isEqual( nextProps.drawChart, prevState.drawChart ) ) {
-			state = { ...state, drawChart: nextProps.drawChart };
-		}
-
-		if ( ! isEqual( nextProps.getParams, prevState.getParams ) ) {
-			state = { ...state, getParams: nextProps.getParams };
-		}
-
-		if ( nextProps.type !== prevState.type ) {
-			state = { ...state, type: nextProps.type };
-		}
+	constructor( props ) {
+		super( props );
 
-		if ( ! isEmpty( state ) ) {
-			return { ...state, params: null };
-		}
-
-		return null;
+		this.chartRef = createRef();
 	}
 
 	componentDidMount() {
-		window.addEventListener( 'resize', this.updateParams );
+		window.addEventListener( 'resize', this.drawChart );
```

Comment:
`this.drawChart` is not defined, should be `this.props.drawChart`?

On second thought. The whole point of this listener was to change the dimensions on resize, but now with those values explicitly defined in props, it appears to no longer be needed.

That responsibility has been shifted up one component:

https://github.com/woocommerce/wc-admin/blob/78fb70a233261b721599e6700dba4719bf4127ac/client/components/chart/index.js#L103

Is that right?

Reply:
You're absolutely right! I like that we can remove those listeners from there. :)

https://github.com/woocommerce/wc-admin/pull/961/commits/925e6ed7a03bb8a11204f3024514a63cc8473c06